crates/interledger-settlement-engines/src/engines/ethereum_ledger/eth_engine.rs

```diff
@@ -292,30 +288,39 @@ where
             .block_number()
             .map_err(move |err| error!("Could not fetch current block number {:?}", err))
             .and_then(move |current_block| {
-                trace!("Current block {}", current_block);
                 // get the safe number of blocks to avoid reorgs
                 let fetch_until = current_block - confirmations;
                 // U256 does not implement IntoFuture so we must wrap it
                 Ok((Ok(fetch_until), store.load_recently_observed_block()))
             })
             .flatten()
-            .and_then(move |(fetch_until, last_observed_block)| {
-                trace!(
-                    "Will fetch txs from block {} until {}",
-                    last_observed_block,
-                    fetch_until
-                );
+            .and_then(move |(to_block, last_observed_block)| {
+                // If we are just starting up, fetch only the most recent block
+                // Note this means we will ignore transactions that were received before
+                // the first time the settlement engine was started.
+                let from_block = if let Some(last_observed_block) = last_observed_block {
+                    if to_block == last_observed_block {
+                        // We already processed the latest block
+                        return Either::A(ok(()));
+                    } else {
+                        last_observed_block + 1
```

Comment:
Good catch, thanks for this fix.

Reply:
It was doing a lot of duplicated queries. This was one reason but I also found that the interval for checking transactions in the example was shorter than the time it took to get get the response to the query and process it, meaning that it would send duplicate queries because the last observed block hadn't been updated yet. That was one reason not to spawn [here](https://github.com/dora-gt/interledger-rs/pull/2/files#diff-c4e39aae2ef778d84a957a8887d4fd03L249)